packages/rpc/src/client.ts

```diff
@@ -27,46 +31,66 @@ import type {
 	SetUserVoiceSettingsData,
 	RelationShip,
 } from './typings/structs';
-import { RPCCommands, RPCEvents, type LobbyType, RelationshipType } from './typings/types';
-import { pid as getPid, uuid } from './util';
-
-function subKey(event: string, args?: unknown[]) {
-	return `${event}${JSON.stringify(args)}`;
-}
+import { RPCCommands, RPCEvents, type LobbyType, RelationshipType, RPCErrorCodes } from './typings/types.js';
+import { pid as getPid } from './util.js';
 
 type Class<T extends new (...args: any[]) => unknown> = T extends new (...args: any[]) => infer R ? R : never;
 
 export interface RPCClientOptions {
-	transport?: keyof typeof transports;
 	origin?: string;
+	/**
+	 * RPC transport. one of `ipc` or `websocket`
+	 */
+	transport?: keyof typeof transports;
 }
 
 export interface RPCLoginOptions {
+	/**
+	 * Access token
+	 */
+	accessToken: string;
+	/**
+	 * Client ID
+	 */
 	clientId: string;
+	/**
+	 * Client secret
+	 */
 	clientSecret: string;
-	accessToken: string;
+	/**
+	 * Should we use RPC token or npt
+	 */
 	rpcToken: string;
```

Comment:
This doesn't look right either

Reply:
It's actually right, because rpc token is obtained from discord, not we supply it

Reply:
Why is the type `string` then? Should it not be a boolean?

Reply:
because typing is weird, and the way that we assign the rpcToken required us to make it as `boolean | string`

i can fix the typing and the naming

https://discord.com/developers/docs/topics/rpc#authorize